pandas/core/indexes/multi.py

```diff
@@ -1304,14 +1304,13 @@ def _get_grouper_for_level(self, mapper, level):
             codes[mask] = ok_codes
             codes[~mask] = -1
 
-        if len(uniques) < len(level_index):
-            # Remove unobserved levels from level_index
-            level_index = level_index.take(uniques)
+            grouper = level_index.take(codes, fill_value=True)
+        else:
+            if len(uniques) < len(level_index):
+                # Remove unobserved levels from level_index
+                level_index = level_index.take(uniques)
 
-        if len(level_index):
```

Comment:
not what i meant

can u eliminate this branch here

Reply:
@jreback 

Ok. I eliminate that branch. But I don't know how not to use branch, do not raise exception together.

Reply:
@jreback 

Ok. I eliminate branch. But I don't know how not to use branch and not conflict with test cases together. Could you give me a clue?

Reply:
@jreback i think the blocker here is clarifying for this contributor what you're asking for

Reply:
I don't think CI logs are around but do we know why this change fails CI? I think @jreback point is that this seems unnecessary so unclear why it is causing the failures

Reply:
@WillAyd 
Because  "Index.take"
```
            if allow_fill and fill_value is not None:
                msg = "Unable to fill values because {0} cannot contain NA"
                raise ValueError(msg.format(self.__class__.__name__))
```
cause fail. 
Some Indexes use ".take" from "Index.take". If `grouper = level_index.take(codes, fill_value=True)` all the time and "level_index" use "take" from "Index.take", then raise exception. 
So I changed check "level_index" empty or not.  this check avoids from exception.

Reply:
Hmm OK. I guess that goes back to this line:

https://github.com/pandas-dev/pandas/blob/2470690b9f0826a8feb426927694fa3500c3e8d2/pandas/core/indexes/base.py#L804

So I guess we would have to allow `fill_value` being passed to Index types that can't hold NA for this to work.

I think I'd be OK with that but any objections @jreback?

```python
>>> pd.Index(range(5)).take([3], fill_value=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 901, in take
    raise ValueError(msg.format(self.__class__.__name__))
ValueError: Unable to fill values because RangeIndex cannot contain NA

>>> pd.Index(range(5)).astype(float).take([3], fill_value=1)
Float64Index([3.0], dtype='float64')
```

Reply:
@WillAyd 
You are right. If Some level index in "MultiIndex", which can't hold NA do not pass `fll_value`. So there is need to check.

Reply:
@proost I think you can replace the `if` check to be `if level_index._can_hold_na`: to at least be more explicit